pkg/controller/clone-controller.go

```diff
@@ -0,0 +1,264 @@
+package controller
+
+import (
+	"fmt"
+	"time"
+	//"encoding/json"  temporary for debug
+	"github.com/golang/glog"
+	. "github.com/kubevirt/containerized-data-importer/pkg/common"
+	"github.com/pkg/errors"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+)
+
+const (
+	// pvc annotations
+	AnnCloneRequest    = "k8s.io/CloneRequest"
+	AnnCloneOf         = "k8s.io/CloneOf"
+	AnnCloneInProgress = "cdi.kubevirt.io/storage.clone.inprogress"
+	
+)
+
+type CloneController struct {
+	clientset                kubernetes.Interface
+	pvcQueue, podQueue       workqueue.RateLimitingInterface
+	pvcInformer, podInformer cache.SharedIndexInformer
+	cloneImage               string
+	pullPolicy               string // Options: IfNotPresent, Always, or Never
+	verbose                  string // verbose levels: 1, 2, ...
+}
+
+func NewCloneController(client kubernetes.Interface, pvcInformer, podInformer cache.SharedIndexInformer, cloneImage string, pullPolicy string, verbose string) *CloneController {
+	c := &CloneController{
+		clientset:     client,
+		pvcQueue:      workqueue.NewRateLimitingQueue(workqueue.DefaultControllerRateLimiter()),
```

Comment:
we should only be using the pvcQueue working queue.

The pod informer's add/update/del functions should be queuing the corresponding pvc's key into the work queue. When the pvc work queue is executed, we look up the status of the corresponding pods from the cache and determine the PVC's status from that.